src/main/java/dev/unnm3d/rediseconomy/command/BackupRestoreCommand.java

```diff
@@ -37,20 +39,51 @@ public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command
             Path userPath = Path.of(plugin.getDataFolder().getAbsolutePath(), args[0]);
             switch (label) {
                 case "backup-economy" -> {
-                    try (FileWriter fw = new FileWriter(userPath.normalize().toFile())) {
-                        StringBuilder sb = new StringBuilder();
-                        currenciesManager.getCurrencies().forEach(currency ->
-                                currency.getAccounts().forEach((uuid, balance) ->
-                                        sb.append(currency.getCurrencyName())
-                                                .append(";")
-                                                .append(uuid.toString())
-                                                .append(";")
-                                                .append(currenciesManager.getUsernameFromUUIDCache(uuid))
-                                                .append(";")
-                                                .append(balance)
-                                                .append(System.getProperty("line.separator"))
-                                ));
-                        fw.write(sb.toString());// currency;uuid;name;balance
+                    final Map<UUID, String> names = new HashMap<>();
+                    for (Map.Entry<String, UUID> entry : currenciesManager.getNameUniqueIds().entrySet()) {
+                        names.put(entry.getValue(), entry.getKey());
+                    }
+
+                    try (BufferedWriter writer = new BufferedWriter(new FileWriter(userPath.normalize().toFile(), true))) {
+                        long memory = Math.min(Runtime.getRuntime().freeMemory(), MAX_MEMORY_USAGE);
+                        int chunkSize = (int) (memory / MEMORY_CHUNK_USAGE) / LINE_SIZE_ESTIMATE;
+
+                        for (Currency currency : currenciesManager.getCurrencies()) {
+                            Map<UUID, Double> accounts = currency.getAccounts();
+                            plugin.getLogger().info("[" + currency.getCurrencyName() + "] Total accounts: " + accounts.size());
+                            if (accounts.isEmpty()) {
```

Comment:
Ok that's fair. but check empty balances too. by default Redis will set them to %starting_balance%